arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARRoutePlannerViewController.swift

```diff
@@ -0,0 +1,240 @@
+    let routeTask = AGSRouteTask(url: URL(string: "https://route.arcgis.com/arcgis/rest/services/World/Route/NAServer/Route_World")!)
+    /// The parameters for route task to solve a route.
+    var routeParameters: AGSRouteParameters!
+    /// The data source to track device location and provide updates to location display.
+    let locationDataSource = AGSCLLocationDataSource()
+    /// A graphic overlay for start and destination graphics.
+    let stopGraphicsOverlay = AGSGraphicsOverlay()
+    
+    /// A graphic overlay for route graphics.
+    let routeGraphicsOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+        overlay.renderer = AGSSimpleRenderer(
+            symbol: AGSSimpleLineSymbol(style: .solid, color: .yellow, width: 5)
+        )
+        return overlay
+    }()
+    
+    /// An OAuth2 configuration for accessing online routing service.
+    let oAuthConfiguration = AGSOAuthConfiguration(
+        portalURL: URL(string: "https://www.arcgis.com")!,
+        clientID: "lgAdHkYZYlwwfAhC",
+        redirectURL: "my-ags-app://auth"
+    )
+    
+    /// An `AGSPoint` representing the start of navigation.
+    var startPoint: AGSPoint? {
+        didSet {
+            resetButtonItem.isEnabled = true
+            let stopSymbol = AGSPictureMarkerSymbol(image: UIImage(named: "StopA")!)
+            let startStopGraphic = AGSGraphic(geometry: self.startPoint, symbol: stopSymbol)
+            stopGraphicsOverlay.graphics.add(startStopGraphic)
+        }
+    }
+    
+    /// An `AGSPoint` representing the destination of navigation.
+    var endPoint: AGSPoint? {
+        didSet {
+            let stopSymbol = AGSPictureMarkerSymbol(image: UIImage(named: "StopB")!)
+            let endStopGraphic = AGSGraphic(geometry: self.endPoint, symbol: stopSymbol)
+            stopGraphicsOverlay.graphics.add(endStopGraphic)
+        }
+    }
+    
+    /// The route result solved by the route task.
+    var routeResult: AGSRouteResult? {
+        willSet(newValue) {
+            if newValue != nil {
+                navigateButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    // MARK: Instance methods
+    
+    /// Create the start and destination stops for the navigation.
+    ///
+    /// - Returns: An array of `AGSStop` objects.
+    func makeStops() -> [AGSStop] {
+        let stop1 = AGSStop(point: self.startPoint!)
+        stop1.name = "Start"
+        let stop2 = AGSStop(point: self.endPoint!)
+        stop2.name = "Destination"
+        return [stop1, stop2]
+    }
+    
+    /// A wrapper function for operations after the route is solved by an `AGSRouteTask`.
+    ///
+    /// - Parameters:
+    ///   - routeResult: The result from `AGSRouteTask.solveRoute(with:completion:)`.
+    ///   - error: The error from `AGSRouteTask.solveRoute(with:completion:)`.
+    func didSolveRoute(with routeResult: AGSRouteResult?, error: Error?) {
```

Comment:
You might consider replacing both parameters with a single `Result` parameter.

Reply:
Great suggestion!